contracts/src/PDPService.sol

```diff
@@ -6,12 +6,106 @@ import "@openzeppelin/contracts/access/Ownable.sol";
 contract PDPService is Ownable {
     // Types 
 
+    // TODO: there is more to think about here.
+    // existing libraries (https://github.com/filecoin-project/filecoin-solidity/blob/master/contracts/v0.8/types/CommonTypes.sol#L91)
+    // don't give us any significant functionality so it makes sense to redeclare.
+    // There will be performance benefits both for storage and memory ops to using bytes32 but we should wait 
+    // on measurment before making that decision.  We could potentially use bytes32 hash and bytes8 for the prefix.
+    struct Cid {
+        bytes data;
+    }
+
     // State fields
+
+    /*
+    A proof set is the metadata required for tracking data for proof of possession.
+    It maintains a list of CIDs of data to be proven and metadata needed to 
+    add and remove data to the set and prove possession efficiently.
+
+    ** logical structure of the proof set**
+    /* 
+    struct ProofSet {
+        Cid[] roots; 
+        uint256[] sizes;
+        uint256[] sumTree;
+        uint256 size;
+        bool deleted;
+    }
+    ** PDP service contract tracks many possible proof sets **
+    []ProofSet proofsets
+
+    To implement this logical structure in the solidity data model we have
+    two arrays tracking the singleton fields and three two dimensional arrays
+    tracking the growing data of the proof set.  The first index is the proof set id
+    and the second index is the index of the data in the array.
+
+    Invariant: rootCids.length == rootSizes.length == sumTreeSizes.length == proofSetSize.length
+    */
+
     uint256 public challengeFinality;
+    // TODO: a single mapping with a pair key i.e. (uint256, uint256 )
+    // might be more efficient as it will half the hashing needed for access and set
+    mapping(uint256 => Cid[]) rootCids; 
+    mapping(uint256 => uint256[]) rootSizes;
+    mapping(uint256 => uint256[]) sumTreeSizes;
```

Comment:
Great points I was mistakenly under different assumptions about how dynamic arrays use slots.  

Since we're moving to singleton I prefer more performance isolation so will keep two mappings for now.